src/Root.cpp

```diff
@@ -607,7 +607,7 @@ bool cRoot::FindAndDoWithPlayer(const AString & a_PlayerName, cPlayerListCallbac
 			size_t Rating = RateCompareString (m_PlayerName, a_pPlayer->GetName());
 			if ((Rating > 0) && (Rating >= m_BestRating))
 			{
-				m_BestMatch = a_pPlayer;
+				m_BestMatch = a_pPlayer->GetName();
```

Comment:
Unless both player have the exact name that is specified in which case the server will just pick one of the players. And it doesn't matter which one, so the fact that this code moves the choice into DoWithPlayer, rather than FindAndDoWithPlayer is irrelevant.

The only scenario in which using the name could fail is if a player name changes whilst the server is running without logging out and back in again, something which I think is not going to happen.